src/renderer/components/Visualiser/VisualiserUtils.js

```diff
@@ -39,20 +39,30 @@ export function limitQuery(query) {
   return limitedQuery;
 }
 
-export function computeAttributes(nodes, graknTx) {
-  return Promise.all(nodes.map(async (node) => {
-    const attributes = (await (await graknTx.query(`match $x id ${node.id}; $x has attribute $y; get $y;`)).collectConcepts());
-    node.attributes = await Promise.all(attributes.map(async (concept) => {
+export async function computeAttributes(nodes, graknTx) {
+  const concepts = await Promise.all(nodes.map(node => graknTx.getConcept(node.id)));
+  const attrIters = await Promise.all(concepts.map(concept => concept.attributes()));
+  const attrGroups = await Promise.all(attrIters.map(iter => iter.collect()));
+
+  return Promise.all(attrGroups.map(async (attrGroup, i) => {
```

Comment:
can we add a comment about what this is doing? Then wouldn't have to read/understand all of the following 20 lines to scan through :)

Reply:
add a comment for the `computeAttributes` function in https://github.com/graknlabs/workbase/pull/282/commits/7a7e13d7f5417a623e0f39170398d085d12a2a5a